uaclient/util.py

```diff
@@ -358,6 +359,34 @@ def get_platform_info() -> 'Dict[str, str]':
     return platform_info
 
 
+def apply_series_overrides(
+        orig_access: 'Dict[str, Any]',
+        copy_dict: bool = True) -> 'Dict[str, Any]':
+    """Return an entitlement dict with an series-specific overrides applied
+
+    :param orig_access: Dict with original entitlement access details
+    :param copy_dict: Boolean set True to copy the orig_access param to avoid
+        altering the original dict.
+    """
+    if not all([isinstance(orig_access, dict), 'entitlement' in orig_access]):
+        raise RuntimeError(
+            'Expected entitlement access dict. Missing "entitlement" key: %s'
+            % orig_access)
+    if copy_dict:
+        updated_access = copy.deepcopy(orig_access)
+    else:
+        updated_access = orig_access
+    series_name = get_platform_info()['series']
+    orig_entitlement = orig_access.get('entitlement', {})
```

Comment:
If we move this up, we can avoid needless `deepcopy` and `get_platform_info` calls in the case we don't have any overrides:

```py
orig_entitlement = orig_access.get('entitlement', {})
overrides = orig_entitlement.get('series', {})
if not overrides:
    return orig_access
```

Reply:
good suggestion will take it.

Reply:
Since we are not going to perform the copy, there is a lot less complexity/cost in this function now. I'll drop this suggestion because our for loop on overrides.items() effectively does no work and doesn't cost much.